Update Zulip tests and re-enable

gnprice opened the pull request:
This has the update to unblock autocomplete changes:
  https://github.com/zulip/zulip-flutter/pull/1190
  https://github.com/flutter/flutter/pull/143249

and with that, tests pass again on Flutter main.

Also install libsqlite3-dev on Linux, a package which is no longer
included in GitHub's new version of the "ubuntu-latest" images.  The
lack of it causes nondeterministic failures when the runner happens
to use the new image:
  https://github.com/flutter/tests/pull/441#issuecomment-2593993199

The actual shared library file libsqlite3.so.0.8.6 -- in the package
libsqlite3-0 -- is still installed by default anyway.  This package
adds the symlink at "libsqlite3.so", which is the filename some
libraries apparently look for.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md

gnprice commented:
Tests pass in Zulip's own CI on Flutter main, after zulip/zulip-flutter#1190, so I expect they'll pass here too. We'll have confirmation on that soon.

@justinmc @Piinks I saw a mention at https://github.com/flutter/tests/pull/437#issuecomment-2588138269 that there was more than one upstream change these tests were blocking. I guess any others haven't yet landed, given that tests pass — what's the other PR (or PRs) involved? (I did just re-scan through Zulip's PR queue and my email, but didn't spot anything.) Happy to take updates as needed.

Piinks commented:
There were a couple PRs that were seeing test flakes actually, @gnprice do you have a sense for how flaky some tests may be in this test suite?

gnprice commented:
We've had a flaky test or two in the past, but I'm not aware of any right now. If you can track down the affected test runs or PRs, I'd be glad to investigate.

Piinks commented:
Other than this comment https://github.com/flutter/tests/pull/437#issuecomment-2588109842
I also heard about it on https://github.com/flutter/flutter/pull/161091/checks which ran into 
```
| ✅ /tmp/flutter_customer_testing.zulip.FCQOYP/tests/test/notifications/display_test.dart: NotificationOpenPayload parse: fails when host is not "notification"
|
| ::error::2120 tests passed, 5 failed, 8 skipped.
ERROR: One or more tests from zulip failed.
Tests finished in 76.33 seconds per iteration.
```

gnprice commented:
Thanks — looking into those.

gnprice commented:
I've investigated and I believe there are no flakes in the tests. There is an infra issue, which caused one and probably both of the failures mentioned above. I'll deal with that.

The failure at https://github.com/flutter/tests/pull/437#issuecomment-2588109842 comes from `libsqlite3.so` being missing on the system. It's currently nondeterministic whether that's present, because this repo's CI uses GitHub's `ubuntu-latest` image, and they're in the middle of a rollout to change what that means:
* https://github.com/actions/runner-images/issues/10636

The failure on https://github.com/flutter/flutter/pull/161091/checks quoted above is hard to pin down because the checks have been re-run since then, and there doesn't seem to be a way to get more details from GitHub (though I spent some time [digging for them](https://discord.com/channels/608014603317936148/692831010445983785/1329185519724269609)). Do you have a more complete log than that snippet?

Lacking more info on that one, though: "5 failed" is exactly the number of tests that would fail for lack of libsqlite3.so, so with no sign of another issue it's probably that one.

Then to actively search out any flakes:
* I've run `flutter test` on the Zulip tree a little over 300 times today, and had zero failures so far.
* I also went through the list in LUCI of failed runs [in Linux customer_testing](https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20customer_testing?limit=200&status=FAILURE) there. I skipped over failures that weren't in any suite (e.g. in downloading dependencies); or that affected multiple suites (e.g. `zulip` but also `super_sliver_list`), figuring that if there were a flake it'd be uncorrelated with other suites, to reduce noise from failed runs that are just a bug in the PR being tested. Over the last month I found 5 PRs that had seen such failures, and all of them appeared to be the test suite doing its job reporting real issues:
  * https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20customer_testing/90791/infra / https://github.com/flutter/flutter/pull/161281/checks
  * https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20customer_testing/90573/infra / https://github.com/flutter/flutter/pull/159013/checks
  * https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20customer_testing/90443/infra / https://github.com/flutter/flutter/pull/160568/checks
  * https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20customer_testing/90170/infra / https://github.com/flutter/flutter/pull/160289/checks
  * and https://github.com/flutter/flutter/pull/143249/checks

gnprice commented:
Fixed that infra issue by installing the package again. Also updated the PR description so it provides an accurate commit message for a squash-merge commit.

Piinks reviewed (approved):
Thanks for the thorough investigation! This LGTM!

justinmc commented:
@gnprice Thanks for jumping on this right after the holidays!

matanlurey commented:
reason for revert: sudo apt-get install does not work on LUCI
